uiowa/uids
# I302 responsive tables

This PR is to work on responsive tables. This is a big thing so i would really like some review if possible If you want to test bot the functionality of tables and their usage in `uids_base`, i suggest starting with the steps in the `To test in uids_base` section.

Browser testing would probably be nice, as I am not sure how many different browsers this has been tested in.

To test in uids:
1. get uids set up and check out this branch
2. run `gulp watch` in the base uids folder
3. check the tables component and make sure that it is working with horizontal and vertical sticky headers. This can be done by making the viewport small and seeing if the headers travel with the viewport.
4. change the table data around so there is only headers in the horizontal, or vertical, and make sure it still works.
5. check that all `<th>` tags are receiving their respective scope tags and that an accessible wrapper is being put around the table.
6. check that you can disable all sticky functionality by adding the `table-static` class to the `table` element
7. check that the sticky top-left corner only works when you have both column AND row headers

To test in uids base:
1. pull down a site using uids base and checkout the branch `i2137-responsive-tables`, open the `package.json` file in `/docroot/themes/uids_base` and under the `dependencies` header, change the `@uiowa/uids` entry to `uiowa/uids#c0c9976`, which should be the most recent commit hash for the branch this is being worked on.
    * (I sometimes had to change the `@uiowa/uids` entry first to `uiowa/uids#i302-responsive-tables` in order for `blt frontend` to grab the `tables.js` file. Once it was in the directory, `uiowa/uids#983f78e` worked. I'm guessing creating a blank `tables.js` file prior to `blt frontend` would also work. - Cory)
2. run a `blt frontend` for it to copy over the branches js and scss and compile them.
3. `blt dsa` a layout builder or paragraphs site and create a new page.
4. make a bunch of sections with differing numbers of columns and place tables in them. you should preferably test tables with both the `.is-striped` class and without it. (ex:  `<table class="table is-striped">` and `<table class="table">`)
5. when viewing the page, all of the tables should function as stated in steps 3-7 from the `To test in uids:` section. As well, they should resize to the column widths when you expand and contract the viewport width.

<details>
<summary>This is some dummy table markup I have been using to make my life easier. Insert it in source editing mode in the WYSIWYG for easy table action!</summary>
<p>

```HTML
<table class="is-striped table">
	<caption>Hello!</caption>
	<thead>
		<tr>
			<th scope="row">Corner</th>
			<th scope="col">Column One</th>
			<th scope="col">Column Two</th>
			<th scope="col">Column Three</th>
			<th scope="col">Column Four</th>
			<th scope="col">Column Five</th>
			<th scope="col">Column Six</th>
			<th scope="col">Column Seven</th>
			<th scope="col">Column Eight</th>
			<th scope="col">Column Nine</th>
		</tr>
	</thead>
	<tbody>
		<tr>
			<th scope="row">Row One</th>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
		</tr>
		<tr>
			<th scope="row">Row Two</th>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
		</tr>
		<tr>
			<th scope="row">Row Three</th>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
		</tr>
		<tr>
			<th scope="row">Row Four</th>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
		</tr>
		<tr>
			<th scope="row">Row Five</th>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
		</tr>
		<tr>
			<th scope="row">Row Six</th>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
		</tr>
		<tr>
			<th scope="row">Row Seven</th>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
		</tr>
		<tr>
			<th scope="row">Row Eight</th>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
		</tr>
		<tr>
			<th scope="row">Row Nine</th>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
		</tr>
		<tr>
			<th scope="row">Row Ten</th>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
			<td>187 Fifth Age</td>
		</tr>
	</tbody>
</table>
